src/AutoRest.CSharp/Mgmt/Decorator/StringExtensions.cs

```diff
@@ -20,5 +21,25 @@ public static string ResourceNameToPlural(this string resourceName)
                 pluralResourceName :
                 $"All{pluralResourceName}";
         }
+
+        /// <summary>
+        /// Add `Resource` suffix to a resource name if that resource doesn't end with `Resource`.
+        /// </summary>
+        /// <param name="resourceName"></param>
+        /// <returns></returns>
+        public static string AddResourceSuffixToResourceName(this string resourceName)
+        {
+            return resourceName.EndsWith(ResourceSuffix) ? resourceName : resourceName + ResourceSuffix;
+        }
+
+        /// <summary>
+        /// Trim `Resource` suffix of a resource name
+        /// </summary>
+        /// <param name="resourceName"></param>
+        /// <returns></returns>
+        public static string TrimResourceSuffixFromResourceName(this string resourceName)
+        {
+            return resourceName.EndsWith(ResourceSuffix) ? resourceName.Substring(0, resourceName.Length - ResourceSuffix.Length) : resourceName;
```

Comment:
In our discussion I thought there was a config allowing you to not do this for things like GenericResource.  Where is that configuration?

Reply:
`request-path-to-resource-name` could do this

Reply:
It's covered in the test project below. Thanks Dapeng for pointing it out.